From 2adfa29406612c5d9716d84ccdea4633621516ed Mon Sep 17 00:00:00 2001
From: Johannes 'josch' Schauer <josch@mister-muffin.de>
Date: Sun, 22 Jan 2017 14:37:33 +0100
Subject: [PATCH 1/5] Install dose-distcheck and lintian for the native
 architecture

When crossbuilding, sbuild will try to install the host architecture
versions of dose-distcheck and lintian. This doesn't make much of a
difference for lintian as that package is Architecture:all right now but
dose-distcheck is Architecture:any and the host architecture version
will not be installable and/or executable. Lets explicitly install the
native version instead.
---
 lib/Sbuild/Build.pm | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/Sbuild/Build.pm b/lib/Sbuild/Build.pm
index 9a767a20..b8a4b402 100644
--- a/lib/Sbuild/Build.pm
+++ b/lib/Sbuild/Build.pm
@@ -1657,7 +1657,7 @@ sub run_lintian {
 	push @lintian_command, $dsc;
     }
 
-    $resolver->add_dependencies('LINTIAN', 'lintian', "", "", "", "", "");
+    $resolver->add_dependencies('LINTIAN', 'lintian:native', "", "", "", "", "");
     return 1 unless $resolver->install_core_deps('lintian', 'LINTIAN');
 
     $session->run_command(
@@ -1885,7 +1885,7 @@ sub explain_bd_uninstallable {
 	# restrictions, we don't want to limit ourselves by it. In cases where
 	# apt cannot find a solution, this check is supposed to allow the user
 	# to know that choosing a different resolver might fix the problem.
-	$resolver->add_dependencies('DOSE3', 'dose-distcheck', "", "", "", "", "");
+	$resolver->add_dependencies('DOSE3', 'dose-distcheck:native', "", "", "", "", "");
 	if (!$resolver->install_core_deps('dose3', 'DOSE3')) {
 	    return 0;
 	}
-- 
2.11.0

